### ISSUE TITLE
Unrelated discussions from HN

### ISSUE THREAD

**Issue by mvolfik** [jacobobryant/hallway#9]
[Hallway page](https://discuss.findka.com/view?url=https%3A%2F%2Fen.wikipedia.org%2Fwiki%2FOhio) for https://en.wikipedia.org/wiki/Ohio shows unrelated HN links. This is issue in the Algolia search, I tried digging into that and what worked for me was surrounding the searched url in quotes (properly escaped, of course):

```python3
requests.get("http://hn.algolia.com/api/v1/search", params={
  "query": '"https://en.wikipedia.org/wiki/Ohio"',
  "restrictSearchableAttributes": "url",
  "attributesToRetrieve": "created_at,num_comments,points,title,author",
  "attributesToHighlight": ""
})
```
The resulting url is  
`https://hn.algolia.com/api/v1/search?query=%22https%3A%2F%2Fen.wikipedia.org%2Fwiki%2FOhio%22&restrictSearchableAttributes=url&attributesToRetrieve=created_at%2Cnum_comments%2Cpoints%2Ctitle%2Cauthor&attributesToHighlight=`

It's also quite helpful to disable highlighting and specify needed attributes, as it reduces the amount of data transferred.

**Comment by mvolfik** [comment]
Except that it kinda doesn't. Good page to test this is `https://en.wikipedia.org/wiki/Go`: [current Hallway](https://discuss.findka.com/view?url=https%3A%2F%2Fen.wikipedia.org%2Fwiki%2FGo) shows many links, the [search as suggested above](https://hn.algolia.com/api/v1/search?query=%22https%3A%2F%2Fen.wikipedia.org%2Fwiki%2FGo%22&restrictSearchableAttributes=url&attributesToRetrieve=created_at%2Cnum_comments%2Cpoints%2Ctitle%2Cauthor&attributesToHighlight=) shows just 2, but there should really be none. [`queryType=prefixNone`](https://www.algolia.com/doc/api-reference/api-parameters/queryType/) sounds like what could work, but it doesn't either

**Comment by jacobobryant** [comment]
Thanks, I've just added this (https://github.com/jacobobryant/hallway/commit/0486b01ab5d9111d348ca47dc77ac04dadae230a). Perhaps `prefixNone` doesn't apply to non-letter characters. e.g. if you search for `Go`, then `prefixNone` would filter out `Gopher` but not `Go.com` or `Go_fever`.

Fortunately, it's easy enough to check if the discussion's URL is an exact match and remove it if not. I just added another commit for that: https://github.com/jacobobryant/hallway/commit/2c9d78d5aad938c79e05e8c8dc0b09ae9ae6428e.
